lib/src/event_arrangers/side_event_arranger.dart

```diff
@@ -16,145 +16,88 @@ class SideEventArranger<T extends Object?> extends EventArranger<T> {
     required double width,
     required double heightPerMinute,
   }) {
-    final durations = _getEventsDuration(events);
-    final tempEvents = [...events]..sort((e1, e2) =>
-        (e1.startTime?.getTotalMinutes ?? 0) -
-        (e2.startTime?.getTotalMinutes ?? 0));
-
-    final table = List.generate(
-      events.length,
-      (index) => List.generate(
-        durations.length,
-        (index) => null as CalendarEventData<T>?, // ignore: unnecessary_cast
-      ),
+    final mergedEvents = MergeEventArranger<T>().arrange(
+      events: events,
+      height: height,
+      width: width,
+      heightPerMinute: heightPerMinute,
     );
 
-    var eventCounter = 0;
-    var rowCounter = 0;
+    final arrangedEvents = <OrganizedCalendarEventData<T>>[];
 
-    while (tempEvents.isNotEmpty && rowCounter < events.length) {
-      eventCounter = 0;
+    for (final event in mergedEvents) {
+      // If there is only one event in list that means, there
+      // is no simultaneous events.
+      if (event.events.length == 1) {
+        arrangedEvents.add(event);
+        continue;
+      }
+
+      final concurrentEvents = event.events;
 
-      var end = tempEvents[0].endTime?.getTotalMinutes ?? 0;
+      if (concurrentEvents.isEmpty) continue;
 
-      _insertIntoTable(table, durations, rowCounter, tempEvents[0]);
+      var column = 1;
+      final sideEventData = <_SideEventData<T>>[];
+      var currentEventIndex = 0;
 
-      tempEvents.removeAt(0);
+      while (concurrentEvents.isNotEmpty) {
+        final event = concurrentEvents[currentEventIndex];
+        final end = event.endTime!.getTotalMinutes;
```

Comment:
handle null check

Reply:
There will be no events with null endTime. As we are already removing events with null endTime or startTime in MergeEvent Arranger at line no 19.